hpx/lcos/local/latch.hpp

```diff
@@ -134,12 +138,9 @@ namespace hpx { namespace lcos { namespace local
         ///
         void wait() const
         {
-            if (counter_.load(std::memory_order_acquire) > 0)
-            {
-                std::unique_lock<mutex_type> l(mtx_.data_);
-                if (counter_.load(std::memory_order_relaxed) > 0)
-                    cond_.data_.wait(l, "hpx::local::latch::wait");
-            }
+            std::unique_lock<mutex_type> l(mtx_.data_);
```

Comment:
Can we do a double-test still? I.e. once before grabbing the lock, just testing the counter and then, if necessary grab the lock and perform a repeated test under the lock?

Reply:
Hmm, don't think so. Just testing `counter_ > 0` outside the lock would get us to the same situation as before. Testing `counter_ > 0 || !notified` is significantly less racey, but still racey. Do you mean something else?

Reply:
Makes sense, thanks!